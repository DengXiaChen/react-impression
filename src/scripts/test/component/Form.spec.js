import React from 'react';
import { shallow, mount } from 'enzyme';
import { Form } from '../../components/impression';

const setup = () => {
    const props = {
        type: 'horizontal',
    };

    const wrapper = shallow(
        <Form {...props}>
            <Form.Group col={6}>Form</Form.Group>
            <Form.Group col={6}>Form</Form.Group>
        </Form>
    );

    const wrapper1 = mount(
        <Form {...props}>
            <Form.Group col={6}>Form</Form.Group>
            <Form.Group col={6}>Form</Form.Group>
        </Form>
    );

    return {
        props,
        wrapper,
        wrapper1,
    };
};

describe('Form', () => {
    const { wrapper, wrapper1 } = setup();

    it('render', () => {
        expect(wrapper1.find('.form-horizontal').exists()).toBe(true);
        expect(wrapper1.find('.form-group')).toHaveLength(2);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
