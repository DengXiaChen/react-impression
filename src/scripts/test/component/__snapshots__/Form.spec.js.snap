// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    className="form-horizontal"
>
    <FormGroup
        col={6}
    >
        Form
    </FormGroup>
    <FormGroup
        col={6}
    >
        Form
    </FormGroup>
</form>,
  "nodes": Array [
    <form
      className="form-horizontal"
>
      <FormGroup
            col={6}
      >
            Form
      </FormGroup>
      <FormGroup
            col={6}
      >
            Form
      </FormGroup>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Form
        grid={false}
        type="horizontal"
>
        <FormGroup
                col={6}
        >
                Form
        </FormGroup>
        <FormGroup
                col={6}
        >
                Form
        </FormGroup>
</Form>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Form {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <FormGroup
              col={6}
>
              Form
</FormGroup>,
            <FormGroup
              col={6}
>
              Form
</FormGroup>,
          ],
          "grid": false,
          "type": "horizontal",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          className="form-horizontal"
>
          <FormGroup
                    col={6}
          >
                    Form
          </FormGroup>
          <FormGroup
                    col={6}
          >
                    Form
          </FormGroup>
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          className="form-horizontal"
>
          <FormGroup
                    col={6}
          >
                    Form
          </FormGroup>
          <FormGroup
                    col={6}
          >
                    Form
          </FormGroup>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Form
    grid={false}
    type="horizontal"
>
    <FormGroup
        col={6}
    >
        Form
    </FormGroup>
    <FormGroup
        col={6}
    >
        Form
    </FormGroup>
</Form>,
}
`;
